editoast/src/client/stdcm_search_env_commands.rs

```diff
@@ -54,6 +57,39 @@ where
     Ok(())
 }
 
+fn parse_active_perimeter(file_path: Option<PathBuf>) -> Option<Json<geos::geojson::Geometry>> {
+    file_path.map(|perimeter_path| {
+        let perimeter_file = File::open(perimeter_path).expect("Perimeter file must exist");
+        let geometry: geos::geojson::Geometry =
+            serde_json::from_reader(BufReader::new(perimeter_file))
+                .expect("Perimeter file can't be read");
+        Json(geometry)
+    })
+}
+
+fn parse_speed_limit_tags(
+    speed_limit_tags: Option<Vec<String>>,
+) -> Option<Json<HashMap<String, i64>>> {
+    speed_limit_tags.map(|tags| {
+        Json(
+            tags.iter()
+                .map(|tag| {
+                    let keyvalue: Vec<&str> = tag.split("|").collect();
+                    if keyvalue.len() != 2 {
+                        panic!("Can't parse speed limit tag, format is MA100|100: {}", tag);
+                    }
```

Comment:
You should be able to use [`split_once`](https://doc.rust-lang.org/std/primitive.str.html#method.split_once) instead. Something along these lines.


```suggestion
                    let Some(key, value) = tag.split_once('|') else {
                        panic!("Can't parse speed limit tag, format is MA100|100: {}", tag);
                    }
```

Reply:
Done